src/scripts/main.js

```diff
@@ -1,3 +1,227 @@
+      const newRow = document.createElement('tr');
+
+      Object.values(obj).forEach((data, index) => {
+        const cell = document.createElement('td');
+
+        cell.textContent =
+          index === 4 && typeof data === 'number'
+            ? `$${data.toLocaleString('en-US')}`
+            : data;
+        newRow.appendChild(cell);
+      });
+
+      tbody.appendChild(newRow);
+    });
+  }
+}
+
+function setActiveRow(cell) {
+  const oldTarget = document.querySelector('.active');
+
+  if (oldTarget) {
+    oldTarget.classList.remove('active');
+  }
+
+  cell.parentElement.classList.add('active');
+}
+
+function createForm() {
+  const form = document.createElement('form');
+
+  form.classList.add('new-employee-form');
+
+  function createLabeledInput(inputName, inputType) {
+    const label = document.createElement('label');
+    const input = document.createElement('input');
+
+    input.name = inputName;
+    input.type = inputType;
+    input.setAttribute('data-qa', inputName);
+    label.textContent = inputName[0].toUpperCase() + inputName.slice(1) + ':';
+    label.appendChild(input);
+
+    return label;
+  }
+
+  const labelName = createLabeledInput('name', 'text');
+
+  form.appendChild(labelName);
+
+  const labelPosition = createLabeledInput('position', 'text');
+
+  form.appendChild(labelPosition);
+
+  const labelOffice = document.createElement('label');
+
+  labelOffice.textContent = 'Office:';
+
+  const selectOffice = document.createElement('select');
+
+  selectOffice.name = 'office';
+  selectOffice.setAttribute('data-qa', 'office');
+
+  const options = [
+    { value: 'Tokyo', text: 'Tokyo' },
+    { value: 'Singapore', text: 'Singapore' },
+    { value: 'London', text: 'London' },
+    { value: 'New York', text: 'New York' },
+    { value: 'Edinburgh', text: 'Edinburgh' },
+    { value: 'San Francisco', text: 'San Francisco' },
+  ];
+
+  options.forEach((option) => {
+    const optionElement = document.createElement('option');
+
+    optionElement.value = option.value;
+    optionElement.textContent = option.text;
+    selectOffice.appendChild(optionElement);
+  });
+
+  labelOffice.appendChild(selectOffice);
+  form.appendChild(labelOffice);
+
+  const labelAge = createLabeledInput('age', 'number');
+
+  form.appendChild(labelAge);
+
+  const labelSalary = createLabeledInput('salary', 'number');
+
+  form.appendChild(labelSalary);
+
+  const submitButton = document.createElement('button');
+
+  submitButton.type = 'submit';
+  submitButton.textContent = 'Save to table';
+  form.appendChild(submitButton);
+
+  document.body.appendChild(form);
+
+  form.addEventListener('submit', (ev) => {
+    ev.preventDefault();
+
+    const userName = form.elements['name'].value.trim();
+    const position = form.elements['position'].value.trim();
+    const office = form.elements['office'].value.trim();
+    const age = parseInt(form.elements['age'].value, 10);
+    const salary = parseFloat(form.elements['salary'].value);
```

Comment:
The `parseFloat` function may return `NaN` if the input is not a valid number. Consider adding a check to ensure the salary is a valid number before proceeding.